x-pack/plugins/alerts/server/invalidate_pending_api_keys/task.ts

```diff
@@ -0,0 +1,193 @@
+  };
+};
+
+export function initializeApiKeyInvalidator(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>,
+  taskManager: TaskManagerSetupContract,
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  registerApiKeyInvalitorTaskDefinition(
+    logger,
+    coreStartServices,
+    taskManager,
+    config,
+    securityPluginSetup
+  );
+}
+
+export async function scheduleApiKeyInvalidatorTask(
+  logger: Logger,
+  config: Promise<AlertsConfig>,
+  taskManager: TaskManagerStartContract
+) {
+  const interval = (await config).invalidateApiKeysTask.interval;
+  try {
+    await taskManager.ensureScheduled({
+      id: TASK_ID,
+      taskType: TASK_TYPE,
+      schedule: {
+        interval,
+      },
+      state: {},
+      params: {},
+    });
+  } catch (e) {
+    logger.debug(`Error scheduling task, received ${e.message}`);
+  }
+}
+
+function registerApiKeyInvalitorTaskDefinition(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>,
+  taskManager: TaskManagerSetupContract,
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  taskManager.registerTaskDefinitions({
+    [TASK_TYPE]: {
+      title: 'Invalidate alert API Keys',
+      createTaskRunner: taskRunner(logger, coreStartServices, config, securityPluginSetup),
+    },
+  });
+}
+
+function taskRunner(
+  logger: Logger,
+  coreStartServices: Promise<[CoreStart, AlertingPluginsStart, unknown]>,
+  config: Promise<AlertsConfig>,
+  securityPluginSetup?: SecurityPluginSetup
+) {
+  return ({ taskInstance }: RunContext) => {
+    const { state } = taskInstance;
+    return {
+      async run() {
+        let totalInvalidated = 0;
+        const configResult = await config;
+        try {
+          const [{ savedObjects }] = await coreStartServices;
+          const repository = savedObjects.createInternalRepository([
+            'api_key_pending_invalidation',
+          ]);
+          const configuredDelay = await configResult.invalidateApiKeysTask.removalDelay;
+          const delay = timePeriodBeforeDate(new Date(), configuredDelay).toISOString();
+
+          let hasApiKeysPendingInvalidation = true;
+          const PAGE_SIZE = 100;
+          do {
+            const apiKeysToInvalidate = await repository.find<InvalidatePendingApiKey>({
+              type: 'api_key_pending_invalidation',
+              filter: `api_key_pending_invalidation.attributes.createdAt <= "${delay}"`,
+              page: 1,
+              sortField: 'createdAt',
+              sortOrder: 'asc',
+              perPage: PAGE_SIZE,
+            });
+            totalInvalidated += await manageToInvalidateApiKeys(
+              logger,
+              repository,
+              apiKeysToInvalidate,
+              securityPluginSetup
+            );
+
+            hasApiKeysPendingInvalidation = apiKeysToInvalidate.total > 0;
```

Comment:
I think this should be PAGE_SIZE since we already invalidated that amount in this iteration?

```suggestion
            hasApiKeysPendingInvalidation = apiKeysToInvalidate.total > PAGE_SIZE;
```